lib/parsers/wav_parser.rb

```diff
@@ -20,43 +20,39 @@ def call(io)
     # with the exception that the Format chunk must precede the Data chunk.
     # The specification does not require the Format chunk to be the first chunk
     # after the RIFF header.
-    # http://soundfile.sapp.org/doc/WaveFormat/
-    # For WAVE files containing PCM audio format we parse the 'fmt ' and
-    # 'data' chunks while for non PCM audio formats the 'fmt ' and 'fact'
-    # chunks. In the latter case the order fo appearence of the chunks is
-    # arbitrary.
-    fmt_processed = false
-    fact_processed = false
+    # https://www.mmsp.ece.mcgill.ca/Documents/AudioFormats/WAVE/WAVE.html
     fmt_data = {}
-    total_sample_frames = 0
+    data_size = 0
+    total_sample_frames = nil
     loop do
       chunk_type, chunk_size = safe_read(io, 8).unpack('a4l')
       case chunk_type
       when 'fmt ' # watch out: the chunk ID of the format chunk ends with a space
         fmt_data = unpack_fmt_chunk(io, chunk_size)
-        return process_non_pcm(fmt_data, total_sample_frames) if fmt_data[:audio_format] != 1 and fact_processed
```

Comment:
We no longer consider audio_format at all. Is that pretty much the fix we implemented?

Reply:
Indeed. The `fact` chunk is _kinda_ mandatory for non-PCM files, but the spec doesn't say it cannot be included in PCM files. Moreover, many non-PCM files don't have the `fact` chunk (the main issue we started this with). So yes, we'll be ignoring the `audio_format` and instead try to make the most of whatever chunks are provided.